src/main/java/org/jabref/gui/importer/actions/GUIPostOpenAction.java

```diff
@@ -33,5 +34,5 @@ public interface GUIPostOpenAction {
      *
      * @param pr    The result of the BIB parse operation.
      */
-    void performAction(ParserResult pr, DialogService dialogService);
+    void performAction(ParserResult pr, DialogService dialogService, PreferencesService preferencesService);
```

Comment:
Can't the injector be used "downstream"?

Reply:
With a proper DI framework, yes. In the next iteration step